ql/ql/src/queries/style/NoUppercaseVariables.ql

```diff
@@ -0,0 +1,42 @@
+/**
+ * @name No upper case variables
+ * @description Variables/fields/predicates should be lower-case, classes/modules should be upper-case
+ * @kind problem
+ * @problem.severity error
+ * @id ql/no-upper-case-variables
+ * @tags correctness
+ * @precision very-high
+ */
+
+import ql
+import codeql_ql.style.AcronymsShouldBeCamelCaseQuery as AcronymsQuery
+
+predicate shouldBeUpperCase(AstNode node, string name, string kind) {
+  name = AcronymsQuery::getName(node, kind) and
+  kind = ["class", "newtypeBranch", "newtype", "module"]
+}
+
+predicate shouldBeLowerCase(AstNode node, string name, string kind) {
+  name = AcronymsQuery::getName(node, kind) and
+  not shouldBeUpperCase(node, name, kind)
+}
+
+string prettyKind(string kind) {
+  exists(string prettyLower | prettyLower = AcronymsQuery::prettyPluralKind(kind) |
+    result = prettyLower.prefix(1).toUpperCase() + prettyLower.suffix(1)
+  )
+}
+
+from string name, AstNode node, string message, string kind
+where
+  (
+    shouldBeLowerCase(node, name, kind) and
+    name.regexpMatch("[A-Z].*") and
```

Comment:
```suggestion
    name.regexpMatch("[^a-z].*") and
```

This will ensure that if any other character appears (underscore, unicode), now or in the future, we'll be alerted. I suggest the same change below.